packages/@okta/vuepress-site/code/java/spring_security_saml/index.md

```diff
@@ -7,217 +7,372 @@ excerpt: >-
+               <button id="logout" type="submit">Logout</button>
+           </form>
+       </body>
+   </html>
+   ```
 
-```bash
-/Library/Tomcat/bin/startup.sh
-```
+1. Create a `src/main/resources/application.yml` file to contain your metadata URI.
 
-**7. Start the Application**
+   ```yaml
+   spring:
+     security:
+       saml2:
+         relyingparty:
+           registration:
+             okta:
+               assertingparty:
+                 metadata-uri: <your-metadata-uri>
+   ```
 
-Load the Spring SAML application by opening this link: `http://localhost:8080/spring-security-saml2-sample/saml/discovery?entityID=http%3A%2F%2Flocalhost%3A8080%2Fspring-security-saml2-sample%2Fsaml%2Fmetadata&returnIDParam=idp`
+### Run the app and authenticate
 
-> **Note:** Links in the app aren't functional yet because we haven't configured any IDPs. Full app functionality is completed after the **Configure Spring Security SAML to work with Okta** section.
+1. Run your Spring Boot app. You can do this from your IDE, or as follows using the command line:
 
-Here's what it should look like: ![App Running](/img/spring-security-saml-intro.png "App Running")
+   ```shell
+   ./gradlew bootRun
+   ```
 
-## Configure Okta to work with Spring Security SAML
+1. Open `http://localhost:8080` in your favorite browser and sign in with the credentials you used to create your account. You should see a successful result in your browser.
+
+   <div class="three-quarter border">
+
+   ![Login success](/img/saml/spring-security-saml-login.png)
+
+   </div>
+
+If you try to sign out, it won't work. You'll fix that in the next section.
+
+### Add a sign out feature
+
+Spring Security's SAML support has a [sign out feature](https://docs.spring.io/spring-security/reference/servlet/saml2/logout.html) that requires configuration.
+
+1. Edit your application on Okta and navigate to **General** > **SAML Settings** > **Edit**.
+
+1. Continue to the **Configure SAML** step and **Show Advanced Settings**. Before you can enable single logout, you'll have to create and upload a certificate to sign the outgoing logout request.
+
+1. You can create a private key and certificate using OpenSSL. Answer at least one of the questions with a value, and it should work.
+
+   ```shell
+   openssl req -newkey rsa:2048 -nodes -keyout local.key -x509 -days 365 -out local.crt
+   ```
+
+1. Copy the generated files to your app's `src/main/resources` directory. Configure `signing` and `singlelogout` in `application.yml`:
+
+   ```yaml
+   spring:
+     security:
+       saml2:
+         relyingparty:
+           registration:
+             okta:
+               signing:
+                 credentials:
+                   - private-key-location: classpath:local.key
+                     certificate-location: classpath:local.crt
+               singlelogout:
+                 binding: POST
+                 response-url: "{baseUrl}/logout/saml2/slo"
```

Comment:
What is {baseUrl}? Is it ${yourOktaDomain}  or is it localhost:8080, in which case you might as well use it here rather than {baseUrl} as you've used it everywhere else.

Reply:
No, it's a placeholder value that Spring Security uses. This is the raw value that you should use and it should not be substituted with anything.

Reply:
OK, thanks for the clarification. Again, new to Spring. Please leave as is.